### PR TITLE
metadata in refTxn

### PR DESCRIPTION
**Pull request by apoorv-2204**
Fixes: Add more informations in reference transaction #86 
Hash and size is calculated before compression.
size in kb
Manual-Test:

Screenshot_20221202_224415

### CODE REVIEW

File: commands/cli.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import path from 'path'
 import fs from 'fs'
+const {
+  createHash
+} = await import('node:crypto');
```
Comment (samuelmanzanera) [inline commands/cli.js:5]: why not using native crypto module ?

File: commands/cli.js [hunk 2]
```diff
@@ -78,5 +78,9 @@ function handleDirectory(entry, files) {
...
   const data = fs.readFileSync(path)
-  files.push({ path, data })
-}
\ No newline at end of file
+  const size = formatBytes(Buffer.byteLength(data));
+  const hash = createHash('md5').update(data).digest('hex');
+  files.push({ path, data, hash, size });
+}
+
+function formatBytes(a, b = 2) { if (!+a) return "0 Bytes"; const c = 0 > b ? 0 : b, d = Math.floor(Math.log(a) / Math.log(1024)); return `${parseFloat((a / Math.pow(1024, d)).toFixed(c))} ${["Bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"][d]}` }
```
Comment (samuelmanzanera) [inline commands/cli.js:86]: a big of indentation would help :wink:

**Comment by apoorv-2204** [comment]
@samuelmanzanera @Neylix 
old
Screenshot_20221129_152957

new
Screenshot_20221129_153735

**Review by bchamagne — COMMENTED** [review]
I tried with this structure: https://github.com/archethic-foundation/archethic-node/files/10056869/myaewebdir.zip
and I get this error message: 

```
[bastien@T495 aeweb-cli]$ ./aeweb.js deploy --seed bastien --endpoint http://127.0.0.1:4000 --path /tmp/myaewebdir
Connecting to http://127.0.0.1:4000
Creating file structure and compress content...
Creating transactions ...
TypeError: Cannot read properties of undefined (reading 'push')
```

I believe the issue comes from the 10MB /dir1/sample-mp4-file.mp4

**Review by bchamagne — APPROVED** [review]

**Comment by apoorv-2204** [comment]
> I tried with this structure: https://github.com/archethic-foundation/archethic-node/files/10056869/myaewebdir.zip and I get this error message:
> 
> ```
> [bastien@T495 aeweb-cli]$ ./aeweb.js deploy --seed bastien --endpoint http://127.0.0.1:4000 --path /tmp/myaewebdir
> Connecting to http://127.0.0.1:4000
> Creating file structure and compress content...
> Creating transactions ...
> TypeError: Cannot read properties of undefined (reading 'push')
> ```
> 
> I believe the issue comes from the 10MB /dir1/sample-mp4-file.mp4

Hi , thanks alot.
I found it was a typo issue.
You may use this branch for updated aeweb hosting.
https://github.com/apoorv-2204/aenode/tree/new_aeweb_hosting


It works well for archethic website.
Screenshot_20221129_204638
Screenshot_20221129_204803

**Comment by apoorv-2204** [comment]
Screenshot_20221130_134647

**Review by bchamagne — APPROVED** [review]

**Review by bchamagne — APPROVED** [review]

**Pull request closed without merging**